index.js

```diff
@@ -162,25 +186,67 @@ class Thermostat {
     return this.api.pushUpdate(this);
   }
 
-  getCurrentHeatingCoolingState(callback) {
+  getActiveState(callback) {
     (async () => {
       await this.api.getThermostats();
-      this.api.log(this.name, 'heating / cooling state: ' + this.powerOn);
+
+      if (this.modletOffline) {
+        return callback(new Error('Modlet broken.'));
+      }
+
+      this.api.log(this.name, 'heating / cooling active: ' + this.powerOn);
+
       callback(null, this.powerOn ?
-        Characteristic.CurrentHeatingCoolingState.COOL :
-        Characteristic.CurrentHeatingCoolingState.OFF);
+        Characteristic.Active.ACTIVE:
+        Characteristic.Active.INACTIVE);
     })();
   }
 
-  setTargetHeatingCoolingState(value, callback) {
-    this.api.log(this.name, 'target heating / cooling state: ' + value);
-    this.powerOn = value === Characteristic.CurrentHeatingCoolingState.COOL;
-    this.update().then(() => callback(null, value));
+  setActiveState(value, callback) {
+    (async () => {
+      await this.api.getThermostats();
+
+      if (this.modletOffline) {
+        return callback(new Error('Modlet broken.'));
+      }
+
+      this.api.log(this.name, 'set heating / cooling active: ' + !this.powerOn);
```

Comment:
setActiveState() is not necessarily always called to toggle, it can be called with the existing value. Use value instead of !this.powerOn